OpenEdXMobile/src/main/java/org/edx/mobile/http/ErrorHandlingOkCallback.java

```diff
@@ -0,0 +1,222 @@
+     *                    error message to deliver.
+     * @param progressCallback The callback to invoke on start and finish of the request. Note that
+     *                         since no callback method in this class is invoked upon request
+     *                         initiation, it assumes that it's being initiated immediately, and
+     *                         thus invokes that start callback immediately as well.
+     */
+    public ErrorHandlingOkCallback(@NonNull final Context context,
+                                   @NonNull final CallTrigger callTrigger,
+                                   @Nullable final TaskProgressCallback progressCallback) {
+        this(context, callTrigger,
+                progressCallback,
+                context instanceof TaskMessageCallback ? (TaskMessageCallback) context : null);
+    }
+
+    /**
+     * Create a new instance of this class.
+     *
+     * @param context A Context for resolving the error message strings. Note that for convenience,
+     *                this will be checked to determine whether it's implementing the
+     *                {@link TaskProgressCallback} interface, and will be registered as such if so.
+     *                If this is not the desired outcome, then the other constructor should be used
+     *                that takes this callback parameter, and it should be explicitly set as null.
+     * @param callTrigger The trigger for initiating the call. This is used to determine the type of
+     *                    error message to deliver.
+     * @param messageCallback The callback to invoke for delivering any error messages.
+     */
+    public ErrorHandlingOkCallback(@NonNull final Context context,
+                                   @NonNull final CallTrigger callTrigger,
+                                   @Nullable final TaskMessageCallback messageCallback) {
+        this(context, callTrigger,
+                context instanceof TaskProgressCallback ? (TaskProgressCallback) context : null,
+                messageCallback);
+    }
+
+    /**
+     * Create a new instance of this class.
+     *
+     * @param context A Context for resolving the error message strings.
+     * @param callTrigger The trigger for initiating the call. This is used to determine the type of
+     *                    error message to deliver.
+     * @param progressCallback The callback to invoke on start and finish of the request. Note that
+     *                         since no callback method in this class is invoked upon request
+     *                         initiation, it assumes that it's being initiated immediately, and
+     *                         thus invokes that start callback immediately as well.
+     * @param messageCallback The callback to invoke for delivering any error messages.
+     */
+    public ErrorHandlingOkCallback(@NonNull final Context context,
```

Comment:
I usually prefer to define constructors in ascending order of the number of parameters, but in this case I agree, and have changed this in commit 5390f039ad77fa398b37138f41df468e486d0948.